content/blog/lift-and-transform-apps-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,283 @@
+
+![vVol Datastore Volumes](/img/vvol-screen-shot-2022-06-14-at-1.12.59-pm.png "vVol Datastore Volumes")
+
+The 64GB volume represents the `/var` filesystem subject for transforming into a Kubernetes persistent volume.
+
+# Iterate and learn
+
+In this next phase, all configuration and testing are performed directly from the Kubernetes cluster. An admin needs to install the HPE CSI Driver for Kubernetes, create a `Secret` that represents the backend user or tenant and setup a `StorageClass` designed to allow importing of volumes.
+
+There are several ways to [deploy the HPE CSI Driver](https://scod.hpedev.io/csi_driver/deployment.html) but the most common method is to use Helm and the gist of it is the following.
+
+```markdown
+$ kubectl create ns hpe-storage
+$ helm install my-hpe-csi-driver hpe-storage/hpe-csi-driver --version 2.1.1-0 -n hpe-storage
+```
+
+Next create a `Secret` for the backend. 
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: hpe-backend
+  namespace: hpe-storage
+stringData:
+  serviceName: nimble-csp-svc
+  servicePort: "8080"
+  backend: <Your Nimble array>
+  username: <Your username with privileges to the vVol folder>
+  password: <Your password>
+```
+
+A `StorageClass` needs to be specifically crafted to allow users to import volumes and snapshots from the storage array.
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: hpe-transform
+provisioner: csi.hpe.com
+parameters:
+  csi.storage.k8s.io/fstype: xfs
+  csi.storage.k8s.io/controller-expand-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-expand-secret-namespace: hpe-storage
+  csi.storage.k8s.io/controller-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  description: Volume transformed from vVol by HPE CSI Driver
+  allowOverrides: importVolAsClone,importVolumeName,forceImport
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+The important details in the `StorageClass` are the `allowOverrides`. Here we put the parameters we want the application admin to override once testing has started (`importVolAsClone`) and once the database is ready for the final import (`importVolumeName` and `forceImport`).
+
+For the final production instance, we’re going to use the [Bitnami MariaDB Helm chart](https://artifacthub.io/packages/helm/bitnami/mariadb). For this chart to install cleanly and be able to bring up the database, the following considerations needs to be taken into account:
```

Comment:
In the last sentence, "needs" should be "need", since it refers to "considerations" which is plural.